Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1

```diff
@@ -90,31 +90,365 @@ Describe 'Identity Cli Functional Tests' {
             }
         }
     }
-    
-    Describe 'Get-ClientRegistration' {}
-        
-    Describe 'New-ClientRegistration' {}
-    
-    Describe 'New-ImplicitClientRegistration' {}
-    
-    Describe 'New-HybridClientRegistration' {}
-    
-    Describe 'New-HybridPkceClientRegistration' {}
-    
-    Describe 'Invoke-UpdateClientRegistration' {}
-    
-    Describe 'Invoke-UpdateClientPassword' {}
-    
-    Describe 'Test-IsClientRegistered' {}
-    
+
+    Describe 'Client Flows' {
+        Context 'Happy Client Lifecycle' {
+            It 'Test, create, update, get, recreate, reset, test' {
+                $timeString = (Get-Date).ToString("hhmmss")
+                $testClientId = "CredentialsHappy$timeString"
+                $fabricToken = Get-FabricInstallerAccessToken -identityUrl $identityUrl -secret $installerSecret
+
+                # Test-IsClientRegistered (expect false)
+                $clientExists = Test-IsClientRegistered -identityUrl $identityUrl -clientId $testClientId -accessToken $fabricToken
+
+                $clientExists | Should -Be $false
+
+                # New-ClientCredentialsClientBody (expect object)
+                $newClient = New-ClientCredentialsClientBody `
+                    -clientId $testClientId `
+                    -clientName "Name for $testClientId" `
+                    -allowedScopes @("fabric/identity.manageresources", "fabric/authorization.read", "fabric/authorization.write", "fabric/authorization.manageclients")
+
+                # New-ClientRegistration (expect client secret)
+                $jsonClient = $newClient | ConvertTo-Json
+                $originalSecret = New-ClientRegistration -identityUrl $identityUrl -body $jsonClient -accessToken $fabricToken
+
+                $originalSecret | Should -Not -Be null
+
+                # Edit-ClientRegistration (expect ok)
+                $newClient["clientName"] = "new Name"
+                $newClient["allowedGrantTypes"] = @("client_credentials", "delegation")
+                $jsonClient = $newClient | ConvertTo-Json
+
+                $response = Edit-ClientRegistration -identityUrl $identityUrl -body $jsonClient -accessToken $fabricToken
+
+                $response | Should -Be ""
+
+                # Get-ClientRegistration (check new name)
+                $updateClient = Get-ClientRegistration -identityUrl $identityUrl -clientId $testClientId -accessToken $fabricToken
+
+                $updateClient | Should -Not -Be null
+
+                $updateClient.clientName | Should -Be "new Name"
+                $updateClient.allowedGrantTypes.Count | Should -Be 2
+
+                # Reset-ClientPassword (expect new secret)
+                $resetSecret = Reset-ClientPassword -identityUrl $identityUrl -clientId $testClientId -accessToken $fabricToken
+                $resetSecret | Should -Not -Be $originalSecret
```

Comment:
Should we also assert that $resetSecret is not null, not just != the original?

Reply:
yes